extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/ticket/TicketValidationService.java

```diff
@@ -75,6 +84,10 @@ public String processUsername(String ticket) throws GuacamoleException {
         catch (TicketValidationException e) {
             throw new GuacamoleException("Ticket validation failed.", e);
         }
+        catch (Throwable t) {
+            logger.error("Error validating ticket with CAS server: {}", t.getMessage());
+            throw new GuacamoleInsufficientCredentialsException("Error validating ticket with CAS server.", t, CredentialsInfo.USERNAME_PASSWORD);
```

Comment:
Okay, I can make the error more generic.  However, throwing GuacamoleInvalidCredentialsException does not display the red "Invalid Login" error message when the username/password field is displayed - I think it has something to do with the fact that the CAS authentication redirects away from the Guacamole page and then back to it.  So, unless there's some trick to getting that to display, I have to stick with the InsufficientCredentials exception.